src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1024,3 +1073,480 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         df._reader = self
         set_api_call_source(df, f"DataFrameReader.{format.lower()}")
         return df
+
+    @private_preview(version="1.29.0")
+    def dbapi(
+        self,
+        create_connection: Callable[[], "Connection"],
```

Comment:
Do we accept a connection object as well?

Reply:
we don't, one reason is that single connection object will have issues with multi-process, Adam can add more reasons